spec/unit/resource/catalog_spec.rb

```diff
@@ -339,6 +340,32 @@
       expect(@catalog.resource("notify[one]", nil)).to equal(@one)
     end
 
+    it "adds resources before an existing resource" do
+      @catalog.add_resource(@one)
+      @catalog.add_resource_before(@one, @two, @three)
+
+      expect(@catalog.resources).to eq([@two, @three, @one])
+    end
+
+    it "raises if adding a resource before a resource not in the catalog" do
+      expect {
+        @catalog.add_resource_before(@one, @two)
+      }.to raise_error(ArgumentError, "Cannot add resource Notify[two] before Notify[one] because Notify[one] is not yet in the catalog")
+    end
+
+    it "adds resources after an existing resource in reverse order" do
+      @catalog.add_resource(@one)
+      @catalog.add_resource_after(@one, @two, @three)
+
+      expect(@catalog.resources).to eq([@one, @three, @two])
```

Comment:
aside from this PR, no one calls `add_resource_after` with multiple args, so we may want to switch the behavior of that method so it preserves the order of the args you pass to it, e.g. returns `[@one, @two, @three]`